control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -423,165 +423,153 @@ func SetupGatewayControllerWithManager(ctx context.Context, mgr ctrl.Manager, co
 		Owns(&corev1.Service{}).
 		Owns(&corev1.Pod{}).
 		Watches(
-			source.NewKindWithCache(&gwv1beta1.ReferenceGrant{}, mgr.GetCache()),
-			handler.EnqueueRequestsFromMapFunc(r.transformReferenceGrant(ctx)),
+			&gwv1beta1.ReferenceGrant{},
+			handler.EnqueueRequestsFromMapFunc(r.transformReferenceGrant),
```

Comment:
You'll see this pattern repeatedly throughout this PR.

With the upgrade to controller-runtime, the watch just accepts the type and uses the cache implicitly ([code](https://github.com/kubernetes-sigs/controller-runtime/blob/5f8d96b043ef8ccadc544535415395be336630e8/pkg/builder/controller.go#L132-L140)).

You'll also notice that a `context.Context` is provided to the handler at runtime, so we no longer have to pass down a context and create a closure in order to use it.